pages/elementsPage.ts

```diff
@@ -0,0 +1,69 @@
+import { Page, Locator, expect } from '@playwright/test';
+
+export class ElementsPage {
+  private page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async navigateAndClick(subCategoryText: string, buttonText: string): Promise<Locator> {
+    await this.page.goto('/elements');
+    const buttonSubCategory = this.page.locator(`text=${subCategoryText}`);
+    await buttonSubCategory.click();
+    const button = this.page.locator(`text=${buttonText}`);
+    return button;
+  }
+
+  async performDoubleClick(): Promise<void> {
+    const doubleClickBtn = await this.navigateAndClick('Buttons', 'Double Click Me');
+    await doubleClickBtn.click({ clickCount: 2 });
+    const doubleClickMessage = this.page.locator('text=You have done a double click');
```

Comment:
`you can put this selector inside the constructor. In the same way, as you described page